plugins/woocommerce/src/Internal/DataStores/Orders/DataSynchronizer.php

```diff
@@ -451,9 +603,46 @@ public function get_description(): string {
 	 * @param WP_Post $post The deleted post.
 	 */
 	private function handle_deleted_post( $postid, $post ): void {
-		if ( 'shop_order' === $post->post_type && $this->data_sync_is_enabled() ) {
+		global $wpdb;
+
+		if ( 'shop_order' !== $post->post_type && 'shop_order_refund' !== $post->post_type ) {
```

Comment:
I think this should include all post types that is returned by `wc_get_order_types( 'cot-migration' )` and not just shop order and refunds.

Reply:
👍🏻 Good point, I'll change it.

Reply:
Fixed in https://github.com/woocommerce/woocommerce/pull/37050/commits/834675d023dd5bf281351793c91b13cb30288872